--- before/inc/template/src/ClientServerGlue/ServerGlue.cs	2014-05-15 21:56:00.000000000 +0200
+++ patched/inc/template/src/ClientServerGlue/ServerGlue.cs	2014-05-15 22:04:07.000000000 +0200
@@ -26,7 +26,7 @@
 using Ict.Petra.Server.App.Core;
 {#USINGNAMESPACES}
 
-namespace Ict.Petra.Server.App.WebService
+namespace {#WEBSERVICENAMESPACE}
 {
 /// <summary>
 /// this publishes the SOAP web services of OpenPetra.org for module {#TOPLEVELMODULE}
