--- before/OpenPetra.build	2014-05-15 21:56:00.000000000 +0200
+++ patched/OpenPetra.build	2014-05-15 22:04:07.000000000 +0200
@@ -52,7 +52,7 @@
   generateORM: generates Object Relational Mapping routines from petra.xml
     sub items: generateORMCachedTables, generateORMData, generateORMAccess, generateORMReferenceCounts
   generateGlue: generates the interfaces and instantiators, which is the glue between client and server
-  generateWinforms: generates all yaml files in csharp\ICT\Petra\Client
+  generateWinforms: generates all yaml files in csharp\ICT\Petra\Client and csharp\ICT\Petra\Plugins
   generateWinform: generate a single yaml file, compile it and optionally and start the client
        nant generateWinform -D:file=MFinance/Gui/GL/GLBatch.yaml
        nant generateWinform startPetraClient -D:file=MFinance/Gui/GL/GLBatch.yaml
@@ -394,6 +394,7 @@
   
   <!-- generate the files for Ict.Petra.Shared.lib.data, ie. the tables and datasets -->
   <nant inheritall="false" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Shared/Ict.Petra.Shared.build')}" verbose="${verbose}"/>
+  <nant inheritall="false" target="custdependORM" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Plugins/Plugins.build')}" verbose="${verbose}"/>
   
   <echo message="Target generateORMData finished at: ${datetime::now()}" />         
 </target>
@@ -415,6 +416,7 @@
   <if test="${WithWindowsClient or WithLinuxClient}">
     <nant inheritall="true" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Client/Ict.Petra.Client.build')}" verbose="${verbose}"/>
     <nant inheritall="true" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/PetraClient/PetraClient.build')}" verbose="${verbose}"/>
+    <nant inheritall="true" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Plugins/Plugins.build')}" verbose="${verbose}"/>
   </if>
   
   <echo message="Target generateWinforms finished at: ${datetime::now()}" />
