--- before/OpenPetra.build	2014-05-10 15:20:00.000000000 +0200
+++ plugin/OpenPetra.build	2014-05-10 15:19:09.000000000 +0200
@@ -52,7 +52,7 @@
   generateORM: generates Object Relational Mapping routines from petra.xml
     sub items: generateORMCachedTables, generateORMData, generateORMAccess, generateORMReferenceCounts
   generateGlue: generates the interfaces and instantiators, which is the glue between client and server
-  generateWinforms: generates all yaml files in csharp\ICT\Petra\Client
+  generateWinforms: generates all yaml files in csharp\ICT\Petra\Client and csharp\ICT\Petra\Plugins
   generateWinform: generate a single yaml file, compile it and optionally and start the client
        nant generateWinform -D:file=MFinance/Gui/GL/GLBatch.yaml
        nant generateWinform startPetraClient -D:file=MFinance/Gui/GL/GLBatch.yaml
@@ -311,6 +311,7 @@
 <target name="generateORMData">
   <!-- generate the files for Ict.Petra.Shared.lib.data, ie. the tables and datasets -->
   <nant inheritall="false" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Shared/Ict.Petra.Shared.build')}" verbose="${verbose}"/>
+  <nant inheritall="false" target="custdependORM" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Plugins/Plugins.build')}" verbose="${verbose}"/>
 </target>
 
 <target name="generateORMAccess">
@@ -324,6 +325,7 @@
   <if test="${WithWindowsClient or WithLinuxClient}">
     <nant inheritall="true" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Client/Ict.Petra.Client.build')}" verbose="${verbose}"/>
     <nant inheritall="true" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/PetraClient/PetraClient.build')}" verbose="${verbose}"/>
+    <nant inheritall="true" target="custdepend" buildfile="${path::combine(dir.csharp, 'ICT/Petra/Plugins/Plugins.build')}" verbose="${verbose}"/>
   </if>
 </target>
 
