--- before/setup/setup.build	2014-05-15 21:56:00.000000000 +0200
+++ patched/setup/setup.build	2014-05-15 22:04:07.000000000 +0200
@@ -438,6 +438,7 @@
     <delete>
       <fileset>
         <include name="${setupDir}/bin30/Ict.Petra.Client*" />
+        <include name="${setupDir}/bin30/Ict.Petra.Plugins.*.Client.dll" />
         <include name="${setupDir}/bin30/PetraClient*" />
         <include name="${setupDir}/bin30/UINavigation.yml" />
       </fileset>
@@ -448,6 +449,7 @@
     <delete>
       <fileset>
         <exclude name="${setupDir}/bin30/Ict.Petra.Server*.connect.dll" />
+        <include name="${setupDir}/bin30/Ict.Petra.Plugins.*.Server.dll" />
         <include name="${setupDir}/bin30/Ict.Petra.Server*" />
         <include name="${setupDir}/bin30/PetraServer*" />
         <include name="${setupDir}/*.sh" />
@@ -494,6 +496,11 @@
          <include name="*.asmx" />
       </fileset>
     </copy>
+    <copy todir="${setupDir}/asmx">
+      <fileset basedir="${dir.root}/csharp/ICT/Petra/Plugins">
+         <include name="**/*.asmx" />
+      </fileset>
+    </copy>
 
     <foreach item="String" property="LinuxTargetDistribution" delim="," in="${LinuxTargetDistribution-list}" >
      <if test="${string::contains(LinuxTargetDistribution, DBMS.Type)}">
