--- openpetraNow.PatchRemoting/inc/template/src/ClientServerGlue/ServerGlue.cs	2014-05-16 08:52:04.000000000 +0200
+++ openpetraNow.PatchPlugin/inc/template/src/ClientServerGlue/ServerGlue.cs	2014-05-16 11:34:44.000000000 +0200
@@ -26,7 +26,7 @@
 using Ict.Petra.Server.App.Core;
 {#USINGNAMESPACES}
 
-namespace Ict.Petra.Server.App.WebService
+namespace {#WEBSERVICENAMESPACE}
 {
 /// <summary>
 /// this publishes the SOAP web services of OpenPetra.org for module {#TOPLEVELMODULE}
