--- a/inc/nanttasks/GenerateProjectFiles.cs	2014-11-17 09:26:30.127483231 +0100
+++ b/inc/nanttasks/GenerateProjectFiles.cs	2014-11-17 09:35:38.319466551 +0100
@@ -266,10 +266,10 @@
                     "Ict.Common,Ict.Petra,Ict.Tools,Ict.Testing");
                 WriteSolutionFile(FTemplateDir, ide.Trim(),
                     "OpenPetra.Server.sln",
-                    "Ict.Common,Ict.Petra.Shared,Ict.Petra.Server,Ict.Petra.PetraServerConsole");
+                    "Ict.Common,Ict.Petra.Shared,Ict.Petra.Server,Ict.Petra.PetraServerConsole,Ict.Petra.Plugins.*.Data,Ict.Petra.Plugins.*.Server");
                 WriteSolutionFile(FTemplateDir, ide.Trim(),
                     "OpenPetra.Client.sln",
-                    "Ict.Common,Ict.Petra.Shared,Ict.Petra.Client,Ict.Petra.PetraClient");
+                    "Ict.Common,Ict.Petra.Shared,Ict.Petra.Client,Ict.Petra.PetraClient,Ict.Petra.Plugins.*.Data,Ict.Petra.Plugins.*.Client");
                 WriteSolutionFile(FTemplateDir, ide.Trim(),
                     "OpenPetra.Tools.sln",
                     "Ict.Common,Ict.Tools");
@@ -447,7 +447,8 @@
 
                 foreach (string incNamespace in IncludeNamespaces)
                 {
-                    if (projectName.StartsWith(incNamespace))
+                    Regex regex = new Regex(incNamespace.Replace(".", "\\.").Replace("*", ".*"));
+                    if (projectName.StartsWith(incNamespace) || regex.Match(projectName).Success)
                     {
                         includeProject = true;
                         break;
